src/Surface.cc

```diff
@@ -63,18 +64,74 @@ class sdf::ODE::Implementation
   public: double slip2 = 0.0;
 };
 
+class sdf::Bullet::Implementation
+{
+  /// \brief The SDF element pointer used during load.
+  public: sdf::ElementPtr sdf{nullptr};
+
+  /// \brief Coefficient of friction in first friction pyramid direction,
+  /// the unitless maximum ratio of force in first friction pyramid
+  /// direction to normal force.
+  public: double friction{1.0};
+
+  /// \brief Coefficient of friction in second friction pyramid direction,
+  /// the unitless maximum ratio of force in second friction pyramid
+  /// direction to normal force.
+  public: double friction2{1.0};
+
+  /// \brief Unit vector specifying first friction pyramid direction in
+  /// collision-fixed reference frame.
+  public: gz::math::Vector3d fdir1{0, 0, 0};
+
+  /// \brief Rolling friction coefficient.
+  public: double rollingFriction{1.0};
+};
+
+class sdf::Torsional::Implementation
+{
+  /// \brief The SDF element pointer used during load.
+  public: sdf::ElementPtr sdf{nullptr};
+
+  /// \brief Torsional friction coefficient. Uunitless maximum ratio of
+  /// tangential stress to normal stress.
+  public: double coefficient{1.0};
+
+  /// \brief If this flag is true, torsional friction is calculated using the
+  /// "patch_radius" parameter. If this flag is set to false,
+  /// "surface_radius" (R) and contact depth (d) are used to compute the patch
+  /// radius as sqrt(R*d).
+  public: bool usePatchRadius{true};
+
+  /// \brief Radius of contact patch surface.
+  public: double patchRadius{0.0};
+
+  /// \brief Surface radius on the point of contact.
+  public: double surfaceRadius{0.0};
+
+  /// \brief Force dependent slip for torsional friction.
+  /// equivalent to inverse of viscous damping coefficient with units of
+  /// rad/s/(Nm). A slip value of 0 is infinitely viscous.
+  public: double odeSlip{0.0};
+};
+
 class sdf::Friction::Implementation
 {
-  /// \brief The object storing contact parameters
+  /// \brief The object storing ode parameters
   public: sdf::ODE ode;
 
+  /// \brief The object storing bullet parameters
+  public: std::optional<sdf::Bullet> bullet;
+
+  /// \brief The object storing torsional parameters
+  public: std::optional<sdf::Torsional> torsional;
```

Comment:
include Tosional and butter sdf headers

Reply:
the classes are defined in `Surface.hh` so included.